src/commands/HBonds.cpp

```diff
@@ -0,0 +1,206 @@
+            throw CFilesError("Useless '--topology-format' without '--topology'");
+        }
+        options.topology_format = args.at("--topology-format").asString();
+    }
+
+    if (args.at("--cell")) {
+        options.custom_cell = true;
+	options.cell = parse_cell(args.at("--cell").asString());
+	}
+
+    options.distance_parameter = 3.0;
+    options.angle_parameter = 30.0;
+    if (args.at("--parameters")) {
+	auto splitted = split(args.at("--parameters").asString(), ':');
+	if (splitted.size() == 2) {
+	    options.distance_parameter = stod(splitted[0]);
+	    options.angle_parameter = stod(splitted[1])*pi/180;
+	} else {
+            throw CFilesError(
+                "custom parameters should be specified as 'd:α'"
+            );
+    	}
+    }	
+
+    return options;
+}
+
+
+std::string HBonds::description() const {
+    return "compute hydrogen bonds network";
+}
+
+int HBonds::run(int argc, const char* argv[]) {
+    auto options = parse_options(argc, argv);
+
+    selectionAcceptor_ = Selection(options.selectionAcceptor);
+    if (selectionAcceptor_.size() != 2) {
+        throw CFilesError("Can not use a selection for acceptors with size different than 2.");
+    }
+
+    selectionDonor_ = Selection(options.selectionDonor);
+    if (selectionDonor_.size() != 1) {
+        throw CFilesError("Can not use a selection for donors with size larger than 1.");
+    }
+
+    auto infile = Trajectory(options.trajectory, 'r', options.format);
+    std::ofstream outfile(options.outfile, std::ios::out);
+    if (outfile.is_open()) {
+        outfile << "#Hydrogen bond network in trajectory " << options.trajectory << std::endl;
+        outfile << "# Selection: acceptors: " << options.selectionAcceptor << " and donors: " << options.selectionDonor << std::endl;
+        outfile << "# Criteria:" << std::endl;
+        outfile << "# donor-acceptor distance < " << options.distance_parameter << " angstroms" << std::endl;
+        outfile << "# donor-acceptor-H angle < " << options.angle_parameter*180/pi << " degrees" << std::endl;
+    } else {
+        throw CFilesError("Could not open the '" + options.outfile + "' file.");
+    }
+
+    if (options.custom_cell) {
+        infile.set_cell(options.cell);
+    }
+
+    if (options.topology != "") {
+        infile.set_topology(options.topology, options.topology_format);
+    }
+
+    for (auto step: options.steps) {
+        if (step >= infile.nsteps()) {
+            break;
+        }
+        auto frame = infile.read_step(step);
+        if (options.guess_bonds) {
+            frame.guess_topology();
+        }
+
+        outfile << "# Frame: " << step << std::endl;        
+
+        auto positions = frame.positions();
+        auto cell = frame.cell();
+
+        auto matched = selectionAcceptor_.evaluate(frame);
+        for (auto match: matched) {
+            assert(match.size() == 2);
+
+            auto acceptor = match[0];
+            auto hydrogen = match[1];
```

Comment:
What happens if the user provides acceptor selection like this: `bonds: name(#1) H and name(#2) O`?